runtime/src/main/java/io/novafoundation/nova/runtime/network/updaters/BlockNumberUpdater.kt

```diff
@@ -1,24 +1,49 @@
 package io.novafoundation.nova.runtime.network.updaters
 
+import android.util.Log
 import io.novafoundation.nova.common.utils.Modules
 import io.novafoundation.nova.common.utils.system
 import io.novafoundation.nova.core.storage.StorageCache
-import io.novafoundation.nova.core.updater.GlobalScope
+import io.novafoundation.nova.core.updater.EmptyScope
+import io.novafoundation.nova.core.updater.SharedRequestsBuilder
+import io.novafoundation.nova.core.updater.UpdateScope
+import io.novafoundation.nova.core.updater.Updater
 import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
-import io.novafoundation.nova.runtime.state.SelectedAssetOptionSharedState
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.getRuntime
 import io.novasama.substrate_sdk_android.runtime.RuntimeSnapshot
 import io.novasama.substrate_sdk_android.runtime.metadata.storage
 import io.novasama.substrate_sdk_android.runtime.metadata.storageKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.onEach
 
 class BlockNumberUpdater(
-    chainRegistry: ChainRegistry,
-    crowdloanSharedState: SelectedAssetOptionSharedState<*>,
-    storageCache: StorageCache
-) : SingleStorageKeyUpdater<Unit>(GlobalScope, crowdloanSharedState, chainRegistry, storageCache) {
+    private val chainRegistry: ChainRegistry,
+    private val storageCache: StorageCache
+) : Updater<Chain> {
 
-    override suspend fun storageKey(runtime: RuntimeSnapshot, scopeValue: Unit): String {
-        return runtime.metadata.system().storage("Number").storageKey()
-    }
+    override val scope: UpdateScope<Chain> = EmptyScope()
```

Comment:
EmptyScope? What is that? Seems like I missed something)
Is it renamed GlobalScope?

Reply:
No this is cope that returns empty flow but typed.

I used it since scope here is unused now

Reply:
Is there a particular reason to not to use GlobalScope?

Reply:
Yes, Global scope is unusable here, since it's typed as Unit but I want to receive Chain in scopeValue in listenForUpdates